src/core/scene/a-scene.js

```diff
@@ -118,8 +124,15 @@ var AScene = module.exports = registerElement('a-scene', {
     addBehavior: {
       value: function (behavior) {
         var behaviors = this.behaviors;
-        if (behaviors.indexOf(behavior) !== -1) { return; }
-        behaviors.push(behavior);
+        for (var b in behaviors) {
```

Comment:
forEach makes me weary as it would create a closure but I'm not really sure how the browsers will optimise it. I went on to make a test case in jsperf but it seems their site is down for maintenance. I'll try again later